cow_instrument/CompositeComponent.cpp

```diff
@@ -30,6 +26,27 @@ void CompositeComponent::shiftPositionBy(const V3D &delta) {
   }
 }
 
+void CompositeComponent::rotate(const Eigen::Vector3d &axis,
+                                const double &theta,
+                                const Eigen::Vector3d &center) {
+  using namespace Eigen;
+  Affine3d transform =
+      Translation3d(center) * AngleAxisd(theta, axis) * Translation3d(-center);
+  Quaterniond rotationPart(transform.rotation());
+  for (size_t i = 0; i < m_children.size(); ++i) {
+
+    m_children[i]->rotate(transform, rotationPart);
+  }
+}
+
+void CompositeComponent::rotate(const Eigen::Affine3d &transform,
+                                const Eigen::Quaterniond &rotationPart) {
+  for (size_t i = 0; i < m_children.size(); ++i) {
+
+    m_children[i]->rotate(transform, rotationPart);
+  }
+}
+
```

Comment:
As just discussed on the whiteboard, we can test the other extreme of having a different parameters for every child component:

```cpp
void CompositeComponent::shiftChildPositionsBy(const std::vector<Vector3d> &deltas) {
  for (size_t i = 0; i < m_children.size(); ++i) {
     m_children[i]->shiftPositionBy(deltas[i]);
   }
}
```

Similarly for rotations.

A new command must be added that contains parameters for **all* pixels. Having a separate command for each pixel would add enormous overhead, since it will trigger a rebuild of, e.g., `DetectorInfo::m_L2s` once for each pixel.